pkg/router/headerformatter.go

```diff
@@ -18,19 +18,21 @@
 package router
 
 import (
+	"context"
+	"mosn.io/mosn/pkg/variable"
 	"strings"
-
-	"mosn.io/mosn/pkg/log"
-	"mosn.io/mosn/pkg/types"
 )
 
 func getHeaderFormatter(value string, append bool) headerFormatter {
-	// TODO: variable headers would be support very soon
-	if strings.Index(value, "%") != -1 {
-		if log.DefaultLogger.GetLogLevel() >= log.WARN {
-			log.DefaultLogger.Warnf("variable headers not support yet, skip, value: %s", value)
+	if len(value) > 2 && strings.HasPrefix(value, "%") && strings.HasSuffix(value, "%") {
+		variableName := strings.Trim(value, "%")
+		// todo cache the variable so we don't need to find it in format method
```

Comment:
delete expired  todo comment.

Reply:
todo comment here is not expired. A better implementation can be that we save the `variable.Check` returned `variable` into the variableHeaderFormatter, so we can just get the value in `format` method. But currently, the `variable` api dosen't provide a proper method to get value from variable, so we use `variable.GetString` method which find value from all registered variables , which can be optimized.

Reply:
oh～ Good idea！